### DISCUSSION TITLE
How to make SASS globally available in my SFCs?

### DISCUSSION THREAD

**Discussion by jader-rubini** [nuxt/nuxt#23067] [Questions]
Hey there folks! I'm taking my first steps into Nuxt, and I can't seem to make my SASS variables, functions, mixins etc work globally within the `style` tag in my SFCs.

I've set up my files so my `@/assets/scss/main.scss` forwards everything else and I just have to reference this one file. So right now my `nuxt.config.js` looks like this:
```
css: [
  'sass',
  'sass-loader',
  '@/assets/sass/main.scss',
],
```
As it is right now, if I manually import this file into my SFC via `@use '@/assets/sass/main.scss' as *` it works just as expected. If I don't, I get an "undefined variable/function/mixin" error. I wouldn't like having to manually import my file into every single SFC, since this looks like repeating and can become problematic as the project scales.

I've read both the Styling section on the Getting Started guide and the Nuxt Config section on the API guide (which is quite vague about the `css` option btw) probably way too many times now, and I still can't figure out what's happening (or not happening, at this case). Can someone help me figure this out?

**Comment by xjccc** [comment]
like this?

```
  vite: {
     css: {
        preprocessorOptions: {
          sass: {
            additionalData: `@import "src/styles/variables.scss";`
          }
        }
      }
  }
```

**Reply by jader-rubini** [reply]
Have tried this already, except I was importing my `main.scss` which is `@forward`ing everything else.
Didn't work neither way, nor does it work if I replace the `@import` rule there with `@use <main.scss> as *`. 😕

**Comment by AloisSeckar** [comment]
In our project, we have `sass` dependency in `package.json`
```
"devDependencies": {
    ...
    "sass": "1.66.1",
    ...
}
```

And then we import global `.scss` files in `<style>` section of `app.vue`:
```
<style lang="scss">
// global styles
@import '../app/assets/scss/cssz-global-base.scss';
@import '../app/assets/scss/cssz-global-colors.scss';
...
</style>
```

I am not aware of any other special configuration related to this.

**Reply by jader-rubini** [reply]
This didn't work either 😔

**Comment by jader-rubini** [comment]
Just for reference, I finally got it working. Maybe this will help someone in the future.

Here's what my `nuxt.config.ts` looks like:
```
[…]
css: ['sass', 'sass-loader', '~/assets/css/main.css'],
vite: {
  css: {
    preprocessorOptions: {
      scss: {
        additionalData: '@use "~/assets/sass/main.scss" as *;',
      },
    },
  },
},
[…]
```
What I was doing wrong:

1. On the `css` config, I'm now only linking to the *compiled CSS* file, as opposed to the source SCSS file. Otherwise I was getting a "module loop" error as the SCSS file was being imported twice, from both the `css` and `vite` config.
2. Also notice the `~` at the beginning of the file paths, as opposed to the `@` I had before. I'm still quite confused about this one, because even in the docs they sometimes use `~` and sometimes `@`. Anyway, from now on in my `nuxt.config.ts` I'll just go with `~`.

Now it works beautifully perfect and I can take full advantage of my functions, variables and mixins right from my SFCs without having to manually import them every time. 🙂

**Reply by Hebilicious** [reply]
This is documented here, is there something that didn't work in the docs? https://nuxt.com/docs/getting-started/styling#using-preprocessors

**Reply by jader-rubini** [reply]
What got me confused is that [here in this section of the docs](https://nuxt.com/docs/api/configuration/nuxt-config#css) about Nuxt Config they use `@`s in the file paths, and they reference both the input and output files, without much explanation.

If I do the same and link both files there: 
1. If I use `@`, I got a 500 with an `undefined mixin` message;
2. If I use `~`, I don't get any errors but my partials won't compile and all my styles relying on them will break.

The only way for it to work 100% is doing like I mentioned above, splitting the config between `css` and `vite`.
